plugin/pkg/shoot/dns/admission.go

```diff
@@ -158,6 +158,15 @@ func (d *DNS) Admit(ctx context.Context, a admission.Attributes, o admission.Obj
 		return err
 	}
 
+	specPath := field.NewPath("spec")
+	// If shoot uses deafult domain validate domain even though shoot can be assigned to seed
+	// having dns disabled
+	if !helper.ShootUsesUnmanagedDNS(shoot) {
+		if err := assignDefaultDomainIfNeeded(a, shoot, d.projectLister, defaultDomains); err != nil {
```

Comment:
Won't this assign a new domain to the Shoot cluster if it doesn't have one? This means that shoot clusters will always have domain names and won't be allowed to be scheduled on a seed that has dns disabled.

Reply:
If the user-specified DNS in shoot yaml doesn't it mean they want to use it. Also if user specifies DNS it can't be scheduled on DNS disabled seed as this validation will fail https://github.com/gardener/gardener/blob/64869baf1aa7cca708dcb174ba59b2549ecad216/plugin/pkg/shoot/dns/admission.go#L229-L231

Reply:
I meant that if the user does not specify a DNS, it will now always get generated. Previously this only happened if the shoot was scheduled on a seed that enables DNS. 
You can even test this out
1. Create a seed with dns disabled
2. Try to create a shoot that has empty DNS setting and schedule it to the seed from 1.
3. See that the shoot creation fails with:
```
k apply -f local/shoot-.yaml 
The Shoot "aws-shoot3" is invalid: spec.dns: Invalid value: core.DNS{Domain:(*string)(0xc005516ef0), Providers:[]core.DNSProvider(nil)}: shoot's .spec.dns section must be nil if seed with disabled DNS is chosen
```

Reply:
@plkokanov PTAL.

Reply:
Looks good now, but I think it will be even better if you split `assignDefaultDomainIfNeeded` into two functions - one that checks if a DNS is valid, and one that assigns DNS to the shoot if it is scheduled on a seed with dns Enabled, but shoot.spec.dns=nil

Reply:
Looks good now, but I think it will be even better if you split `assignDefaultDomainIfNeeded` into two functions - one that checks if a DNS is valid, and one that assigns DNS to the shoot if it is scheduled on a seed with dns Enabled, but shoot.spec.dns=nil

Reply:
@plkokanov PTAL.